Source/KSCrash/Recording/Tools/KSJSONCodec.c

```diff
@@ -1137,8 +1137,19 @@ int ksjsoncodec_i_decodeElement(const char** ptr,
                 return KSJSON_ERROR_INCOMPLETE;
             }
 
+            // our buffer is not necessarily NULL-terminated, so
+            // it would be undefined to call sscanf/sttod etc. directly.
+            // instead we create a temporary string.
             double value;
-            sscanf(start, "%lg", &value);
+            size_t len = (unsigned)(*ptr - start);
+            char * buf = malloc(len + 1);
```

Comment:
I've redone it using `strtod()`, which doesn't exhibit the same bug as `sscanf()`.

I went with malloc because we're using it inside `ksjsoncodec_i_decodeString` — does that also need fixing?